### PR TITLE
Updates anomaly docstrings

### LINKED ISSUE #3536: Update anomaly related docstrings

### PR DESCRIPTION
**Pull request by rhyolight**
WIP

fixes #3536

### CODE REVIEW

**Comment by rhyolight** [comment]
@marionleborgne feel free to review, but technically I need a 👍 from a reviewer to merge.

File: src/nupic/algorithms/anomaly.py [hunk 1]
```diff
@@ -56,48 +56,53 @@ def computeRawAnomalyScore(activeColumns, prevPredictedColumns):
...
 
 
-  def __init__(self, 
-               slidingWindowSize=None, 
-               mode=MODE_PURE, 
+  def __init__(self,
+               slidingWindowSize=None,
+               mode=MODE_PURE,
                binaryAnomalyThreshold=None):
-    """
```
Comment (marionleborgne) [inline src/nupic/algorithms/anomaly.py]: Why did you remove the doc string in the __init__?

Comment (rhyolight) [inline src/nupic/algorithms/anomaly.py]: I have been moving all init docs into the class docs because it works better with Sphinx.

Comment (scottpurdy) [inline src/nupic/algorithms/anomaly.py]: We should have good doc strings for `__init__` functions. Usually the only info you need here is the parameter documentation. But I don't think we should move that to the class doc string just because of docs tooling.

Comment (rhyolight) [inline src/nupic/algorithms/anomaly.py]: @scottpurdy Can I add a docstring in each `__init__` that points to the class definition? This seems like a common convention in python projects using Sphinx.

Comment (scottpurdy) [inline src/nupic/algorithms/anomaly.py]: After looking into the Sphinx issue, I don't have a strong opinion. You might want to check with others since this is something all engineers will have to be aware of. But I'm fine with whatever route you want to go.
